packages/edit-navigation/src/components/header/use-menu-locations.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { useDispatch } from '@wordpress/data';
+
+export default function useMenuLocations() {
+	const [ menuLocationsByName, setMenuLocationsByName ] = useState( null );
+
+	useEffect( () => {
+		const fetchMenuLocationsByName = async () => {
+			const newMenuLocationsByName = await apiFetch( {
+				method: 'GET',
+				path: '/__experimental/menu-locations',
+			} );
+			setMenuLocationsByName( newMenuLocationsByName );
```

Comment:
We will want to prevent setting state if the component is unmounted. A simple fix would be to add a local variable `didMount`.

```js
useEffect( () => {
  let didMount = true;

  // ... async stuffs.
    if (didMount) setMenuLocationsByName( newMenuLocationsByName );
 
  return () => {
    didMount = false;
  };
}, [] );
```

Reply:
Good catch, thanks! Fixed in 26fe95b571.